### DISCUSSION TITLE
Ability to append current params

### DISCUSSION THREAD

**Discussion by ghzb** [tighten/ziggy#640] [Ideas]
#### Description

Some routes specifically those in admin panels, tend to have multiple parameters because of breadcrumb navigation.  I frequently find myself writing code like the following to add a single parameter which is context sensitive to the route.

```js
client.post(
    route('admin.settings.update', {
        ...route().params, 
        model: 1
     })
);
```

#### Suggestion

My suggestion is to provide a method that simplifies the above code. maybe something like this?
```js
client.post(
    route('admin.settings.update', route().params().merge({
        model: 1
     }))
);
```

#### Alternatives

Another approach could be to provide a method for adding a single parameter. This would be more useful for adding one off parameters like filters.
```js
client.post(
    route('admin.settings.update', route().params().extend('model', 1)
);
```

**Comment by bakerkretzmar** [comment]
This is cool, I like the idea of some kind of `merge()` method to add/overwrite partial params to the current set.
